tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -2178,6 +2178,38 @@ def visit_functiondef(self,node):
         except:
             pass
 
+class NoAzureCoreHttpResponseImport(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    """Rule to check that we aren't importing azure.core.pipeline.transport.HttpResponse outside of Azure Core."""
+    name = "no-azure-core-http-response-import"
+    priority = -1
+    msgs = {
+        "C4756": (
+            "Do not import HttpResponse from Azure Core.",
+            "no-azure-core-http-response-import",
+            "Do not import HttpResponse from Azure Core."
+        ),
+    }
+
+    AZURE_CORE_NAME = "azure.core"
+    AZURE_CORE_TRANSPORT_NAME = "azure.core.pipeline.transport"
+    RESPONSE_CLASSES = {"HttpResponse", "AsyncHttpResponse"}
+
+    def visit_importfrom(self, node):
+        """Check that we aren't importing HttpResponse."""
+        if node.root().name.startswith(self.AZURE_CORE_NAME): 
+            return
+        if node.modname == self.AZURE_CORE_TRANSPORT_NAME: 
```

Comment:
Does this check specifically for the `from x import y` type of import? Do we also want to look for usage of `import azure.core.transport.pipeline` / `azure.core.transport.pipeline.HttpResponse` in the code?